bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/LocaleResolver.java

```diff
@@ -0,0 +1,21 @@
+package org.bahmni.module.bahmnicore.web.v1_0;
+
+import org.openmrs.api.APIException;
+import org.openmrs.util.LocaleUtility;
+
+import java.util.Locale;
+
+public class LocaleResolver {
+
+    public static Locale identifyLocale(String locale) {
+        if (locale != null && !locale.isEmpty()) {
+            Locale searchLocale = LocaleUtility.fromSpecification(locale);
+            if (searchLocale.getLanguage().isEmpty()) {
+                throw new APIException("Invalid locale: " + locale);
+            }
+            return searchLocale;
```

Comment:
before returning, should you check for whether this is supported locale? not all locales be supported in an implementation.

Reply:
Do we have all supported locale defined in somewhere ?

Reply:
I am guessing this is ok. It will not return any result worst case. have you tested with a valid but unsupported locale (locale=de_AT) along with request param?